Feature/similar movie screen

baraa0abd opened the pull request:
https://github.com/user-attachments/assets/cf4f905c-8dcc-4ff0-9443-128df13135bd

baraa0abd committed:
Merge remote-tracking branch 'origin/feature/details-top-cast-screen-ui' into feature/similar-movie-screen

baraa0abd committed:
feat: similar movies screen

Added the similar movies screen.

baraa0abd committed:
fix: string resources

Add "Similar Movies" string resource in English and Arabic.
Update SimilarMoviesScreen to use the new string resource.

baraa0abd committed:
Merge branch 'develop' into feature/similar-movie-screen

# Conflicts:
#	presentation/src/main/res/values-ar/string.xml
#	presentation/src/main/res/values/string.xml

baraa0abd committed:
Refactor/similar-movies-screen

- Use Movie domain model
- Remove dummy data
- Clean unused code from PreferredMediaUseCase

fatimafaisalghazi commented on presentation/src/main/java/com/madrid/presentation/screens/similarMovies/SimilarMoviesScreen.kt:
```diff
@@ -0,0 +1,109 @@
+import com.madrid.designsystem.component.MovioIcon
+import com.madrid.designsystem.component.MovioText
+import com.madrid.presentation.composables.movioCards.MovioVerticalCard
+import com.madrid.domain.entity.Movie
```
do we need to use the domain here ?

baraa0abd committed:
feat: Enhance SimilarMoviesScreen with loading/error states

- Integrate `SearchViewModel` to manage state for Similar Movies.
- Display a "Loading..." message while fetching data.
- Show an error message if data fetching fails.
- Update movie click handling to use `SearchScreenState.MovieUiState`.
- Add "loading" string resource in English and Arabic.

fatimafaisalghazi commented on presentation/src/main/java/com/madrid/presentation/screens/similarMovies/SimilarMoviesScreen.kt:
```diff
@@ -0,0 +1,138 @@
+    val state by viewModel.state.collectAsState()
+    var selectedMovieId by remember { mutableStateOf<String?>(null) }
+
+    Column(
+        modifier = modifier
+            .fillMaxSize()
+            .background(AppTheme.colors.surfaceColor.surface)
+    ) {
+        Row(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(horizontal = 16.dp, vertical = 12.dp),
+            verticalAlignment = Alignment.CenterVertically
+        ) {
+            Box(
+                modifier = Modifier
+                    .size(40.dp)
+                    .clickable { onBackClick() },
+                contentAlignment = Alignment.Center
+            ) {
+                MovioIcon(
+                    painter = painterResource(R.drawable.arrow_left),
+                    contentDescription = "",
+                    tint = AppTheme.colors.surfaceColor.onSurface
+                )
+            }
+            Spacer(modifier = Modifier.weight(1f))
+            MovioText(
+                text = stringResource(id = com.madrid.presentation.R.string.selected_similar_movie),
+                textStyle = AppTheme.textStyle.headLine.largeBold18,
+                color = AppTheme.colors.surfaceColor.onSurface
+            )
+            Spacer(modifier = Modifier.weight(1f))
+        }
```
repace it  with TopAppBar

fatimafaisalghazi commented on presentation/src/main/java/com/madrid/presentation/screens/similarMovies/SimilarMoviesScreen.kt:
```diff
@@ -0,0 +1,138 @@
+            Spacer(modifier = Modifier.weight(1f))
+        }
+        LazyVerticalGrid(
+            modifier = Modifier.fillMaxSize(),
```
.statusBarsPadding(),

baraa0abd committed:
Refactor: similar movies screen top app bar

The `SimilarMoviesScreen` now utilizes the `TopAppBar` composable for its header, replacing the previous custom `Row`-based implementation.

The top app bar includes a back arrow icon and the title "Similar Movies". The back arrow's click functionality remains, triggering the `onBackClick` callback. The screen now also incorporates `statusBarsPadding` for better layout consistency.

AmrNasserSaad committed:
Merge remote-tracking branch 'origin/develop' into feature/similar-movie-screen

# Conflicts:
#	presentation/src/main/res/values-ar/string.xml
#	presentation/src/main/res/values/string.xml
